hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/staking/StakingRewardsHelper.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.service.token.impl.handlers.staking;
+
+import static com.hedera.node.app.service.mono.utils.Units.HBARS_TO_TINYBARS;
+import static java.util.Objects.requireNonNull;
+
+import com.hedera.hapi.node.base.AccountAmount;
+import com.hedera.hapi.node.base.AccountID;
+import com.hedera.hapi.node.state.token.Account;
+import com.hedera.node.app.service.token.ReadableAccountStore;
+import com.hedera.node.app.service.token.impl.WritableAccountStore;
+import com.hedera.node.app.service.token.impl.WritableNetworkStakingRewardsStore;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * Helper class for staking reward calculations
+ */
+@Singleton
+public class StakingRewardsHelper {
+    private static final Logger log = LogManager.getLogger(StakingRewardsHelper.class);
+    public static final long MAX_PENDING_REWARDS = 50_000_000_000L * HBARS_TO_TINYBARS;
+
+    @Inject
+    public StakingRewardsHelper() {
+        // Exists for Dagger injection
+    }
```

Comment:
I don't like to see many static methods. Atleast all helper classes imo need not be static.  Since it is easy to inject via Dagger I made the class non static

Reply:
@mhess-swl , In this class `getPossibleRewardReceivers`, `isRewardSituation` and `asAccountAmounts` can be static. I am fine with that.
Technically speaking `decreasePendingRewardsBy` and `increasePendingRewardsBy` methods should exist `stakingRewardStore`. But since those stores are only used for CRUD, I have added to this helper class. 

In my opinion, I use static methods when they are used in multiple classes as utility methods. 
Also, I use static methods that have no side effects, like mutating state etc.,
`decreasePendingRewardsBy` and `increasePendingRewardsBy` are doing that. It is just a personal opinion.

Reply:
I have changed `getPossibleRewardReceivers` and `isRewardSituation` to static methods as suggested.